keras/utils/generic_utils.py

```diff
@@ -9,10 +9,21 @@
 import marshal
 import types as python_types
 
+global_custom_objects = {}
```

Comment:
Thanks @fchollet ! The code currently has that behavior, but it might be worth changing. Seems inconsistent with other usages but I didn't want to change something in case someone was depending on it.

Current keras behavior:
```python
config=...
#This call passes custom_objects
layer_from_config({'class_name':'Qwerty','config':config},{'Qwerty':Dense})
#This call will use the custom_objects from the previous call
layer_from_config({'class_name':'Qwerty','config':config})
```

It is odd behavior but I don't know if anyone is currently depending on it. This PR keeps that behavior, but if someone wants to clear that global cache they can. If the global custom objects is unsafe, is that something you would want to change?

The intent of this PR was not to change that unsafe behavior, but to be able to ameliorate it.

I like your idea of a clean RAII wrapper for help manage the scope.

So between these two directions, which would you prefer? Or is there another option entirely?
* Keep current behavior with custom_objects leaking, but add some scope functions to control it
* Change behavior so custom_objects don't leak, but add some scope functions if you want to share cusom_objects across calls

RE: Contrib
Another thing to think about is the contrib repository. A lot of people are going to be writing objects in other modules and wanting those to load correctly and be created by name. One way to do that would be something like a global custom object registry that modules could add to. That could work with either the scope idea or the global idea.

Cheers,
Ben